frontend/src/components/main-page/groups-tree-page/common/dependencies/dependencies.tsx

```diff
@@ -15,13 +15,14 @@ import { generatePath, useNavigate } from 'react-router-dom';
 
 import { Icons } from '../../../../../icons';
 import { GROUPS_TREE_ROUTES_ABS } from '../../../../../routes';
-import { ServiceDocsServiceTreeItem } from '../../../utils/service-docs-utils';
+import { APINode, EventNode, MainNode } from '../../../service-docs-tree';
+import { getAllAPIsAndEvents } from '../../../utils/service-docs-tree-utils';
 
 import { DependencyDetails } from './dependency-details';
 import { VisualizationModal } from './visualization/visualization-modal';
 
 interface Props {
-  service: ServiceDocsServiceTreeItem;
+  showDependenciesFor: MainNode;
 }
 export const Dependencies: React.FC<Props> = (props) => {
   const controller = useController(props);
```

Comment:
I think the first lines of code below should belong to the component one level above:

```
 <Typography variant="h3">Dependencies</Typography>
      <Box sx={{ marginTop: 2 }}>
        <Button
          variant="outlined"
          onClick={(): void => controller.setShowVisualization(true)}
        >
          Open Visualization
        </Button>
      </Box>
....
```

I think it makes sense to show different headings depending on the user sees the dependencies of a service or a group because presenting the same thing might cause confusion. Thus, I'd call it `Service Dependencies` for services, and `Aggregated Dependencies` for groups to highlight where we currently are.